src/neighbors.cc

```diff
@@ -551,16 +510,13 @@ neighbors_init(void)
 {
     struct servent *sep = nullptr;
     const char *me = getMyHostname();
-    CachePeer *thisPeer = nullptr;
-    CachePeer *next = nullptr;
 
     neighborsRegisterWithCacheManager();
 
     if (Comm::IsConnOpen(icpIncomingConn)) {
+        RawCachePeers peersToRemove;
 
-        for (thisPeer = Config.peers; thisPeer; thisPeer = next) {
-            next = thisPeer->next;
-
+        for (const auto &thisPeer: CurrentCachePeers()) {
```

Comment:
The temporary `peersToRemove` list and extra work can be elided by making this for-loop iterate backwards over the STL container and removing peers as it finds them.

```suggestion
        for (const auto thisPeer = Config.peers->rbegin(); thisPeer++; thisPeer != Config.peers->rend()) {
```

Reply:
Removing elements from a container while iterating over it is not a good idea, since erase() [invalidates](https://en.cppreference.com/w/cpp/container/vector/erase) iterators (e.g., in vectors). And in fact, CachePeers::Storage is a vector.

Reply:
One more reason for it not to be a vector. Also the vector type is why I suggested `rbegin()` instead of a normal forward or ranged-for scan.
Anyway, this is optimization, I do not insist.